lib/abstractivator/proc_ext.rb

```diff
@@ -8,6 +8,12 @@ def loosen_args
       Proc.loose_call(self, args, kws, &block)
     end
   end
+
+  KEYWORD_PARAMETER_TYPES = %i(key keyreq keyrest).freeze
+
+  def accepts_keywords
+    @accepts_keywords ||= parameters.select{|param| KEYWORD_PARAMETER_TYPES.include?(param.first)}.any?
```

Comment:
You can use `Enumerable#any?` with a block here, instead of `select`ing and then calling `any?`.